dvc/repo/experiments/queue/celery.py

```diff
@@ -193,17 +192,24 @@ def _iter_processed(self) -> Generator[_MessageEntry, None, None]:
             entry_dict = kwargs.get("entry_dict", args[0])
             yield _MessageEntry(msg, QueueEntry.from_dict(entry_dict))
 
-    def _iter_active_tasks(self) -> Generator[_TaskEntry, None, None]:
+    @cached_property
+    def _running_id(self) -> Set[str]:
+        running_id: Set[str] = set()
+        for _, status in self.worker_status.items():
+            task = first(status)
+            if task:
+                running_id.add(task["id"])
+        return running_id
 
+    def _iter_active_tasks(self) -> Generator[_TaskEntry, None, None]:
         for msg, entry in self._iter_processed():
             task_id = msg.headers["id"]
             result: AsyncResult = AsyncResult(task_id)
-            if not result.ready():
+            if not result.ready() and task_id in self._running_id:
```

Comment:
This is the right idea, and I think all that is really needed is to be doing something like

```
running_ids = self._get_running_task_ids()
if not result.ready() and task_id in running_ids:
    ...
```
in order to properly handle (not) caching the worker status